modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/CopyFolderMVCRenderCommand.java

```diff
@@ -31,6 +49,61 @@
 )
 public class CopyFolderMVCRenderCommand extends BaseFolderMVCRenderCommand {
 
+	@Override
+	public String render(
+			RenderRequest renderRequest, RenderResponse renderResponse)
+		throws PortletException {
+
+		try {
+			DLFolder dlFolder = _dlFolderLocalService.getDLFolder(
+				ParamUtil.getLong(renderRequest, "sourceFolderId"));
+
+			long dlFolderSize = _dlFolderLocalService.getFolderSize(
+				dlFolder.getCompanyId(), dlFolder.getGroupId(),
+				dlFolder.getTreePath());
+
+			try {
+				_validateCopyToSize(
+					_dlSizeLimitConfigurationProvider.getCompanyMaxSizeToCopy(
+						dlFolder.getCompanyId()),
+					_dlSizeLimitConfigurationProvider.getGroupMaxSizeToCopy(
+						dlFolder.getGroupId()),
+					_dlSizeLimitConfigurationProvider.getSystemMaxSizeToCopy(),
+					dlFolderSize);
+
+				return super.render(renderRequest, renderResponse);
+			}
+			catch (FolderSizeLimitExceededException
+						folderSizeLimitExceededException) {
+
+				HttpServletRequest originalHttpServletRequest =
+					_portal.getOriginalServletRequest(
+						_portal.getHttpServletRequest(renderRequest));
+
+				SessionErrors.add(
+					originalHttpServletRequest.getSession(),
```

Comment:
Why are you passing the session explicitly? Unless there's a specific reason for it, we always use the portletRequest (or httpServletRequest if appropriate).

Reply:
As we discussed early, this solution only works passing the session explicitly.